test/properties/property_test.py

```diff
@@ -0,0 +1,212 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""PropertyTest class"""
+
+from test import QiskitNatureTestCase
+
+import numpy as np
+
+from qiskit_nature.properties.second_quantization.driver_metadata import DriverMetadata
+from qiskit_nature.properties.second_quantization.electronic import (
+    AngularMomentum,
+    DipoleMoment,
+    ElectronicDipoleMoment,
+    ElectronicEnergy,
+    Magnetization,
+    ParticleNumber,
+)
+from qiskit_nature.properties.second_quantization.electronic.bases import ElectronicBasisTransform
+from qiskit_nature.properties.second_quantization.electronic.integrals import (
+    ElectronicIntegrals,
+    OneBodyElectronicIntegrals,
+    TwoBodyElectronicIntegrals,
+)
+from qiskit_nature.properties.second_quantization.vibrational import (
+    OccupiedModals,
+    VibrationalEnergy,
+)
+from qiskit_nature.properties.second_quantization.vibrational.integrals import VibrationalIntegrals
+
+
+class PropertyTest(QiskitNatureTestCase):
+    """Property instance tester"""
+
+    def compare_angular_momentum(
+        self, first: AngularMomentum, second: AngularMomentum, msg: str = None
+    ) -> None:
+        """Compares two AngularMomentum instances."""
+        if first._num_spin_orbitals != second._num_spin_orbitals:
+            raise self.failureException(msg)
+        if first._spin != second._spin:
+            raise self.failureException(msg)
```

Comment:
Can we do anything here (and similar below such as in magnetization and particle number) to again avoid testing against internal private fields? In this case `spin` has a public getter (and setter) property, so that could be used. Should we add getter/setters for the others too so that we would test the tolerances that are passed in too. One alternative, or in addition, that had occurred to me was to add an `__eq__` method in the Property that did the compare across the significant/appropriate fields - presumably this would be the ones that are saved/loaded. If we chose not to expose some fields, then the fact that eq would operate internally off private fields seems fine. I had not mentioned doing an equals before due to potential precision over comparison of arrays so maybe it was not so doable to have equals for any Property - unless it was always done by default at given precision (maybe even overriddable).

Reply:
I did not define `__eq__` directly as we so far avoided doing that within Qiskit Nature. I am not 100% sure of the motivation why this was avoided thus far (probably the precision you mention) but I think for now I would like to finalize this PR instead of diving into that discussion. I will add setters/getters where needed and adapt these test methods accordingly.